tests/testsuite/update.rs

```diff
@@ -2096,3 +2095,341 @@ fn update_breaking_specific_packages() {
         )
         .run();
 }
+
+#[cargo_test]
+fn update_breaking_specific_packages_that_wont_update() {
+    Package::new("compatible", "1.0.0").publish();
+    Package::new("renamed-from", "1.0.0").publish();
+    Package::new("non-semver", "1.0.0").publish();
+    Package::new("bar", "1.0.0")
+        .add_dep(Dependency::new("transitive-compatible", "1.0.0").build())
+        .add_dep(Dependency::new("transitive-incompatible", "1.0.0").build())
+        .publish();
+    Package::new("transitive-compatible", "1.0.0").publish();
+    Package::new("transitive-incompatible", "1.0.0").publish();
+
+    let crate_manifest = r#"
+        # Check if formatting is preserved
+
+        [package]
+        name  =  "foo"
+        version  =  "0.0.1"
+        edition  =  "2015"
+        authors  =  []
+
+        [dependencies]
+        compatible  =  "1.0"  # Comment
+        renamed-to  =  { package  =  "renamed-from", version  =  "1.0" }  # Comment
+        non-semver  =  "~1.0"  # Comment
+        bar  =  "1.0"  # Comment
+    "#;
+
+    let p = project()
+        .file("Cargo.toml", crate_manifest)
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+    let lock_file = p.read_file("Cargo.lock");
+
+    Package::new("compatible", "1.0.1").publish();
+    Package::new("renamed-from", "1.0.1").publish();
+    Package::new("non-semver", "1.0.1").publish();
+    Package::new("transitive-compatible", "1.0.1").publish();
+    Package::new("transitive-incompatible", "1.0.1").publish();
+
+    Package::new("renamed-from", "2.0.0").publish();
+    Package::new("non-semver", "2.0.0").publish();
+    Package::new("transitive-incompatible", "2.0.0").publish();
+
+    p.cargo("update -Zunstable-options --breaking compatible renamed-from non-semver transitive-compatible transitive-incompatible")
```

Comment:
Should this fail?

I lean towards "yes"
- During testing, it will be easier to get feedback on an unwanted failure
- Compatibility wise, its easier to stop erroring than to start

We can split fixing this into a separate PR if you want.

Reply:
Yes, let's do a separate PR.

Reply:
Done. See https://github.com/rust-lang/cargo/issues/12425#issuecomment-2186198258